index.js

```diff
@@ -18,28 +18,44 @@ function isMergeableObject(val) {
 
 function defaultArrayMerge(target, source, optionsArgument) {
     var destination = target.slice()
+    var clone = optionsArgument && optionsArgument.clone === true
     source.forEach(function(e, i) {
         if (typeof destination[i] === 'undefined') {
             destination[i] = e
```

Comment:
I think `e` needs to be cloned in this case, too - I think a failing test for this case might be

```js
var a = { key: 'yup' }
var target = []
var source = [a]
var output = deepmerge(target, source)
t.notEqual(output[0], a)
t.equal(output[0].key, 'yup')
```

But I haven't tested that to make sure it fails right now.